### PR TITLE
cilium-cli 0.8.6 (new formula)

### PR DESCRIPTION
**Pull request by tklauser**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This adds the Cilium CLI to install, manage & troubleshoot Kubernetes
clusters running Cilium. See https://cilium.io for more information.

### CODE REVIEW

File: Formula/cilium-cli.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    bash_output = Utils.safe_popen_read(bin/"cilium", "completion", "bash")
+    (bash_completion/"cilium").write bash_output
+    zsh_output = Utils.safe_popen_read(bin/"cilium", "completion", "zsh")
+    (zsh_completion/"_cilium").write zsh_output
+    fish_output = Utils.safe_popen_read(bin/"cilium", "completion", "fish")
+    (fish_completion/"cilium.fish").write fish_output
+  end
+
+  test do
+    assert_match(/cilium-cli: v#{version}/, shell_output("#{bin}/cilium version"))
```
Comment (carlocab) [inline Formula/cilium-cli.rb:23]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (tklauser) [reply Formula/cilium-cli.rb:23]: Unfortunately, all commands apart from `cilium help` and `cilium version` require a running Kubernetes cluster to work correctly. If that is not the case, they will just report an error message which I considered to be a bad test as well.

Any advice on how to proceed with this? I see some existing formulae also use `version` and/or `help` in their tests.

Comment (SMillerDev) [reply Formula/cilium-cli.rb:23]: From the linked document:

> If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.

Comment (tklauser) [reply Formula/cilium-cli.rb]: Thanks. I've updated the formula to run `cilium install` and `cilium hubble enable` and assert the errors.

File: Formula/cilium-cli.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    (bash_completion/"cilium").write bash_output
+    zsh_output = Utils.safe_popen_read(bin/"cilium", "completion", "zsh")
+    (zsh_completion/"_cilium").write zsh_output
+    fish_output = Utils.safe_popen_read(bin/"cilium", "completion", "fish")
+    (fish_completion/"cilium.fish").write fish_output
+  end
+
+  test do
+    assert_match(/Cluster name "" is not valid/, shell_output("#{bin}/cilium install 2>&1", 1))
+    assert_match(/Error: Unable to enable Hubble/, shell_output("#{bin}/cilium hubble enable 2>&1", 1))
```
Comment (carlocab) [inline Formula/cilium-cli.rb:24]:
```suggestion
    assert_match('Cluster name "" is not valid', shell_output("#{bin}/cilium install 2>&1", 1))
    assert_match("Error: Unable to enable Hubble", shell_output("#{bin}/cilium hubble enable 2>&1", 1))
```
We don't need these to be `Regexp`s.

Comment (tklauser) [reply Formula/cilium-cli.rb:24]: Done, thanks.

**Review by chenrui333 — DISMISSED** [review]

File: Formula/cilium-cli.rb (hunk 2 above)
Comment (carlocab) [reply Formula/cilium-cli.rb:24]: You actually forgot to push this change, I think.

Comment (tklauser) [reply Formula/cilium-cli.rb]: Indeed, thanks. Now pushed.

**Review by SMillerDev — APPROVED** [review]
Thanks @tklauser  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**